app/src/converter.js

```diff
@@ -2,17 +2,52 @@ import open from 'open';
 
 const os = require("os");
 const path = require('path');
+const fs = require("node:fs");
 
 const logger = require('./logger');
 const rpc = require('./weh-rpc');
 
 const exec_dir = path.dirname(process.execPath);
-let ffmpeg = path.join(exec_dir, "ffmpeg");
-let ffprobe = path.join(exec_dir, "ffprobe");
+let ffmpeg = ensureProgramExt(path.join(exec_dir, "ffmpeg"));
+let ffprobe = ensureProgramExt(path.join(exec_dir, "ffprobe"));
```

Comment:
I recommend that here you do something like this:

let ffmpeg = find ExecutableFullPath("ffmpeg", `${exec_dir}:${process.env.PATH}`);

And in `ExecutableFullPath`:

```js
function findExecutableFullPath (programName, paths) {
  if (os.platform() == "win32") {
    programeName += ".exe";
  }
  const foundExecutablePath = paths.split(path.delimiter)
   …
}
```

This will reduce the amount of code.

Reply:
Refactored.